src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs

```diff
@@ -138,410 +144,589 @@ private bool TryGetValueFromDocument(BlittableObjectInstance parent, string prop
 
-            private JsValue TranslateToJs(BlittableObjectInstance owner, string key, BlittableJsonToken type, object value)
+            PropertyDescriptor[] items = new PropertyDescriptor[bjra.Length];
+            for (var i = 0; i < bjra.Length; i++)
             {
-                switch (type & BlittableJsonReaderBase.TypesMask)
+                var json = bjra.GetValueTokenTupleByIndex(i);
+                BlittableJsonToken itemType = json.Item2 & BlittableJsonReaderBase.TypesMask;
+                JsValue item;
+                if (itemType == BlittableJsonToken.Integer || itemType == BlittableJsonToken.LazyNumber)
                 {
-                    case BlittableJsonToken.Null:
-                        return DynamicJsNull.ExplicitNull;
+                    item = TranslateToJs(null, null, json.Item2, json.Item1);
+                }
+                else
+                {
+                    item = TranslateToJs(parent, null, json.Item2, json.Item1);
+                }
 
-                    case BlittableJsonToken.Boolean:
-                        return (bool)value ? JsBoolean.True : JsBoolean.False;
+                items[i] = new PropertyDescriptor(item, true, true, true);
+            }
 
-                    case BlittableJsonToken.Integer:
-                        // TODO: in the future, add [numeric type]TryFormat, when parsing numbers to strings
-                        owner?.RecordNumericFieldType(key, BlittableJsonToken.Integer);
-                        return (long)value;
+            var jsArray = new ArrayInstance(e, items);
+            jsArray.SetPrototypeOf(e.Array.PrototypeObject);
 
-                    case BlittableJsonToken.LazyNumber:
-                        owner?.RecordNumericFieldType(key, BlittableJsonToken.LazyNumber);
-                        return GetJsValueForLazyNumber(owner?.Engine, (LazyNumberValue)value);
+            return jsArray;
+        }
 
-                    case BlittableJsonToken.String:
-                        return value.ToString();
+        private JsValue TranslateToJs(BlittableObjectInstance owner, string key, BlittableJsonToken type, object value)
+        {
+            switch (type & BlittableJsonReaderBase.TypesMask)
+            {
+                case BlittableJsonToken.Null:
+                    return DynamicJsNull.ExplicitNull;
 
-                    case BlittableJsonToken.CompressedString:
-                        return value.ToString();
+                case BlittableJsonToken.Boolean:
+                    return (bool)value ? JsBoolean.True : JsBoolean.False;
 
-                    case BlittableJsonToken.StartObject:
-                        Changed = true;
-                        _parent.MarkChanged();
-                        BlittableJsonReaderObject blittable = (BlittableJsonReaderObject)value;
+                case BlittableJsonToken.Integer:
+                    // TODO: in the future, add [numeric type]TryFormat, when parsing numbers to strings
```

Comment:
Let's talk about this TODO directly. I don't understand

Reply:
It's not my comment, something happened with the formatting and it appeared as a new one. Sorry.